packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts

```diff
@@ -55,44 +54,70 @@ export abstract class BaseDeployExecutor extends SfdxCommandletExecutor<
     });
 
     execution.processExitSubject.subscribe(async exitCode => {
-      const telemetry = new TelemetryBuilder();
-      let success = false;
-      try {
-        BaseDeployExecutor.errorCollection.clear();
-        if (stdOut) {
-          const deployParser = new ForceDeployResultParser(stdOut);
-          const errors = deployParser.getErrors();
-          if (errors && !deployParser.hasConflicts()) {
-            channelService.showChannelOutput();
+      await this.exitProcessHandlerDeploy(
+        exitCode,
+        stdOut,
+        workspacePath,
+        execFilePathOrPaths,
+        execution,
+        startTime,
+        cancellationToken,
+        cancellationTokenSource
+      );
+    });
+  }
+
+  protected async exitProcessHandlerDeploy(
+    exitCode: number | undefined,
+    stdOut: string,
+    workspacePath: string,
+    execFilePathOrPaths: string,
+    execution: CliCommandExecution,
+    startTime: [number, number],
+    cancellationToken: vscode.CancellationToken | undefined,
+    cancellationTokenSource: vscode.CancellationTokenSource
+  ): Promise<void> {
+    if (exitCode === 0 && this.getDeployType() === DeployType.Push) {
+      const pushResult = JSON.parse(stdOut);
+      this.updateCache(pushResult);
+    }
+
+    const telemetry = new TelemetryBuilder();
+    let success = false;
+    try {
+      BaseDeployExecutor.errorCollection.clear();
+      if (stdOut) {
+        const deployParser = new ForceDeployResultParser(stdOut);
+        const errors = deployParser.getErrors();
+        if (errors && !deployParser.hasConflicts()) {
+          channelService.showChannelOutput();
+          if (workspacePath && execFilePathOrPaths) {
             handleDiagnosticErrors(
               errors,
               workspacePath,
               execFilePathOrPaths,
               BaseDeployExecutor.errorCollection
             );
-          } else {
-            success = true;
           }
-          this.outputResult(deployParser);
-        }
-      } catch (e) {
-        BaseDeployExecutor.errorCollection.clear();
-        if (e.name !== 'DeployParserFail') {
-          e.message =
-            'Error while creating diagnostics for vscode problem view.';
+        } else {
+          success = true;
         }
-        telemetryService.sendException(e.name, e.message);
-        console.error(e.message);
+        this.outputResult(deployParser);
```

Comment:
what happens here if we get to the `handleDiagnosticErrors` call above?

Reply:
It looks like handleDiagnosticErrors is called when there are errors/conflicts, and it is supposed to output a table of the conflicts?  But right now it looks like it is just exiting and not showing a table at all:

To set this up, I created a conflict on the server for one file, then changed another one locally so I had two files to push, one with a conflict:

![2023-05-20_06-43-12 (1)](https://github.com/forcedotcom/salesforcedx-vscode/assets/46458081/7c21709e-e569-43d5-8f03-1df5496a644e)